### PR TITLE
Added comprehensive error checking to nplh and testing

### PR DESCRIPTION
**Pull request by fharding1**

### CODE REVIEW

File: nplh.go [hunk 1]
```diff
@@ -24,17 +24,22 @@ func done(msg string) {
...
 
 	if strings.HasPrefix(path, "~/") {
 		path = filepath.Join(dir, path[2:])
 	}
-	return path
+
+	return path, nil
 }
 
-func fileExists(path string) (exists bool) {
```
Comment (calebeby) [inline nplh.go:37]: Why did you remove this?

Comment (fharding1) [reply nplh.go:37]: I didn't... I removed the named return value exists since you aren't using it. I really just modified it.

Comment (calebeby) [reply nplh.go:37]: Why did you remove the named from the return value is what I am asking

Comment (calebeby) [reply nplh.go:37]: Isn't it useful for autocompletion?

Comment (fharding1) [reply nplh.go:37]: It's useful for documentation if you are returning something that is not obvious, but since you aren't using the automatically created named return value and it's obvious that `fileExists` returns whether the file exists, I think it should be removed.

Comment (calebeby) [reply nplh.go:37]: OK, fair
